src/components/Filter/Filter.js

```diff
@@ -0,0 +1,389 @@
+import cx from 'classnames';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { DropdownButton } from 'react-bootstrap';
+import { MenuItem } from '../MenuItem';
+import { bindMethods } from '../../common/helpers';
+
+class Filter extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      currentField: null,
+      currentValue: '',
+      filterCategory: null,
+      filterValue: null,
+    };
+    bindMethods(this, [
+      'selectField',
+      'selectValue',
+      'onValueKeyPress',
+      'updateCurrentValue',
+    ]);
+  }
+
+  setDefaultField(props) {
+    const { currentField } = this.state;
+
+    if (props.fields && props.fields.length && !currentField) {
+      this.setState({
+        currentField: props.fields[0],
+      });
+    }
+  }
+
+  componentWillMount() {
+    this.setDefaultField(this.props);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setDefaultField(nextProps);
+  }
+
+  selectField(field) {
+    const { currentField } = this.state;
+
+    if (field === currentField) {
+      return;
+    }
+
+    var updateState = {
+      currentField: field,
+      currentValue: '',
+    };
+
+    if (field.filterType === 'complex-select') {
+      updateState.filterCategory = field.filterCategories[0];
+    }
+
+    this.setState(updateState);
+  }
+
+  selectValue(value, valueType) {
+    const { filterAddedCB } = this.props;
+    const { currentField, filterCategory } = this.state;
+
+    if (value !== undefined) {
+      if (currentField.filterType === 'complex-select') {
+        switch (valueType) {
+          case 'filter-category':
```

Comment:
Can we pull these literals in the case statements out into a `constants.js` file.? I think it would line up more with the Charts/ToastNotifications etc..